### PR TITLE
Stop scheduled publishing

### PR DESCRIPTION
**Pull request by emmabeynon**
For https://trello.com/c/FndkZ8gG/843-allow-users-to-stop-a-scheduled-publishing

This extracts the existing unschedule controller action into an interactor and creates a dedicated Unschedule Service to take care of making the appropriate status changes to a scheduled edition and deleting the publish intent that had been created when the edition was scheduled.

### CODE REVIEW

File: spec/services/unschedule_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  let(:edition) { create(:edition, :scheduled, scheduling: scheduling) }
+  let(:user) { build(:user) }
+
+  describe "#unschedule" do
+    it "sets an edition's scheduled publishing datetime to nil" do
+      stub_publishing_api_destroy_intent(edition.base_path)
+      UnscheduleService.new(edition).unschedule(user: user)
+
+      expect(edition.scheduled_publishing_datetime).to be nil
+    end
```
Comment (kevindew) [inline spec/services/unschedule_service_spec.rb]: I assume this test is no longer relevant?

File: app/services/unschedule_service.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  def unschedule(user:)
+    set_edition_status(user)
+    destroy_publishing_intent
+  end
+
+private
+
+  def set_edition_status(user)
+    updater = Versioning::RevisionUpdater.new(edition.revision, user)
+    edition.assign_revision(updater.next_revision, user).save!
```
Comment (kevindew) [inline app/services/unschedule_service.rb]: I think this code isn't needed anymore, I assume it's from when we were setting the date to nil?

Comment (emmabeynon) [reply app/services/unschedule_service.rb]: Yes, good catch thank you.

**Comment by emmabeynon** [comment]
Thanks @kevindew.  Makes sense to keep everything in the Unschedule Interactor, especially now that I've taken another couple of lines of code out of it.  I've also added a feature test for when the Publishing API is down.

**Pull request merged**